Assets/Scripts/Combat/CreatureMechanics.cs

```diff
@@ -51,11 +51,13 @@ public void RegisterStatusEffect(StatusEffect effect)
 
     public int BeginTurnAndGetMaxActionPoints()
     {
-        int ap = 5 + GetEffectiveSpeed() / 10;
+        int ap = 5 + GetEffectiveSpeed();
         foreach (StatusEffect effect in statusEffects)
         {
             ap = effect.PerRoundEffect(ap);
         }
+        // If dead from PerRoundEffect (like poison), take no actions.
+        if (dead) return 0;
```

Comment:
I would move this at the top of the method.
If (dead) nothing else needs to be computed.

Reply:
I moved it inside the loop, to make it better clarified why it has to be after PerRoundEffect.